packages/sensors_plus/tizen/src/sensors_plus_plugin.cc

```diff
@@ -67,40 +71,87 @@ class SensorsPlusPlugin : public flutter::Plugin {
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/gyroscope",
             &flutter::StandardMethodCodec::GetInstance());
     gyroscope_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kGyroscope));
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kGyroscope,
+                                                    interval_));
 
     std::unique_ptr<FlEventChannel> user_accel_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/user_accel",
             &flutter::StandardMethodCodec::GetInstance());
     user_accel_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kUserAccel));
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kUserAccel,
+                                                    interval_));
 
     std::unique_ptr<FlEventChannel> magnetometer_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/magnetometer",
             &flutter::StandardMethodCodec::GetInstance());
     magnetometer_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kMagnetometer));
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kMagnetometer,
+                                                    interval_));
   }
+
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    const std::string &method_name = method_call.method_name();
+    const flutter::EncodableValue *arguments = method_call.arguments();
+
+    if (method_name == "setAccelerationSamplingPeriod" ||
+        method_name == "setGyroscopeSamplingPeriod" ||
+        method_name == "setUserAccelerometerSamplingPeriod" ||
+        method_name == "setMagnetometerSamplingPeriod") {
+      const auto *sampling_period = std::get_if<int32_t>(arguments);
+      if (sampling_period) {
+        // TODO(jsuya): In the sensor_plus_platform_interface, the
+        // set***SamplingPeriod method call is called only before creating an
+        // Event channel. Tizen has sensor_listener_set_interval, but it does
+        // not support setting the sampling period after creating an event
+        // channel because there is no use case. This may be supported in the
+        // future.
+        interval_ = *sampling_period / 1000;  // sampling_period is microsecond
+                                              // and interval_ is millisecond.
+      } else {
+        result->Error("Invalid argument", "No sampling period provided.");
+        return;
+      }
+    } else {
+      result->NotImplemented();
+      return;
+    }
+    result->Success();
+  }
+
+  int interval_ = 0;
```

Comment:
According to the documentation, 0 means the system default interval (typically 100 ms). Can we distinguish between the default interval and `SensorInterval.fastestInterval`, which has a value of 0?

Reply:
done